examples/automm/memory_bank/memory_bank.py

```diff
@@ -0,0 +1,484 @@
+    parser.add_argument("--aug_epochs", type=int, default=1, help="The epochs to create the bank.")
+    parser.add_argument("--model_head_type", type=str, default="linear", help="The model head for few-shot classification. Choose from 'linear' and 'SVM'.")
+    parser.add_argument("--lr", type=float, default=1e-2, help="The learning rate for training the model head.")
+    parser.add_argument("--lr_F", type=float, default=1e-3, help="The learning rate for finetuing the memory bank.")
+    parser.add_argument("--train_epoch", type=int, default=20, help="The training epochs for training the model head.")
+    parser.add_argument("--train_epoch_F", type=int, default=20, help="The training epochs for finetuning the memory bank.")
+    parser.add_argument("--init_alpha", type=float, default=1.17, help="The initial value of hyper-parameter alpha in memory bank. Alpha adjusts the weight of probability between the classifier and memory bank.")
+    parser.add_argument("--init_beta", type=float, default=1., help="The initial values of hyper-parameter beta in memory bank. Beta modulates the sharpness when converting the similarities into non-negative values.")
+    parser.add_argument("--search_scale", type=int, nargs="+", default=[10, 10], help="The search scale of alpha and beta.")
+    parser.add_argument("--search_step", type=int, nargs="+", default=[200, 20], help="The steps of searching hyper-parameters alpha and beta.")
+    args = parser.parse_args()
+
+    args.bank_dir = os.path.join('./memory_bank_models', args.dataset)
+    os.makedirs(args.bank_dir, exist_ok=True)
+
+    if args.column_names is None:
+        args.column_names = ["text"] if args.type == "text" else ["image"]
+    
+    if args.backbone is None:
+        if args.type == "text":
+            args.backbone = "sentence-transformers/paraphrase-mpnet-base-v2"
+        elif args.type == "image":
+            args.backbone = "swin_base_patch4_window7_224"
+        else:
+            args.backbone = "openai/clip-vit-large-patch14-336"
+    
+    return args
+
+
+class AutoMMMemoryBank(nn.Module):
+    """
+    The model to generate few shot predict probability with 
+    features extracted by AutoGluon MultiModal Predictor.
+    """
+
+    def __init__(
+        self, 
+        bank_keys, 
+        bank_labels,
+        hidden_size, 
+        num_classes,
+        clip_weights=None, 
+        model_head_type="linear",
+    ):
+        """
+        Create the model head and the memory bank.
+
+        Parameters
+        ----------
+        bank_keys
+            The content of bank composed of features in the training set.
+        bank_labels 
+            The labels of corresponding bank_keys.
+        hidden_size
+            The size of features.
+        num_classes
+            The classes of the dataset.
+        clip_weights
+            The encoded label-text prompts which is only used in type "clip".
```

Comment:
We can change it to `clip embedding of the semantic text that describes the labels`.